processing/data_collection/gazette/spiders/base.py

```diff
@@ -186,3 +186,83 @@ def get_gazette(self, document):
             date=dateparser.parse(document[1], languages=("pt",)).date(),
             file_urls=(document[0],),
         )
+
+
+class DoemGazetteSpider(BaseGazetteSpider):
+    """
+    Base spider for all cities listed on https://doem.org.br
+    """
+
+    allowed_domains = ["doem.org.br"]
+
+    def start_requests(self):
+        yield scrapy.Request(self.get_url(), callback=self.parse_pagination)
+
+    def parse_pagination(self, response):
+        """
+        This parse function is used to get all the pages available and
+        return request object for each one
+        """
+        return [
+            scrapy.Request(self.get_url(page), callback=self.parse)
+            for page in range(1, self.get_last_page(response) + 1)
+        ]
+
+    def parse(self, response):
+        """
+        Parse each page from the results page and yield the gazette issues available.
+        """
+        GAZETTE_CONTAINER_CSS_QUERY = "div.box-diario"
+        gazette_boxes = response.css(GAZETTE_CONTAINER_CSS_QUERY)
+
+        for gazette_box in gazette_boxes:
+            file_url = self.get_pdf_url(gazette_box)
+            date = self.get_gazette_date(gazette_box)
+            edition_number = self.get_edition_number(gazette_box)
+            yield Gazette(
+                date=date,
+                file_urls=[file_url],
+                edition_number=edition_number,
+                power="executive_legislative",
+            )
+
+    def get_url(self, page=1):
+        state, city = self.name.split("_")
+        return f"https://doem.org.br/{state}/{city}/pesquisar?data_inicial=2013-01-01&page={page}"
+
+    @staticmethod
+    def get_last_page(response):
+        """
+        Gets the last page number available in the pages navigation menu
+        """
+        PAGINATION_CSS_QUERY = "ul.pagination li a::text"
+        pages = response.css(PAGINATION_CSS_QUERY).getall()
+        return max([int(page) for page in pages if page.isnumeric()])
```

Comment:
I like small and succinct methods. Really cool! Just a curiosity, why `@staticmethod` ?

Reply:
I saw this being done on a few spiders that I checked for reference, and I assumed that was the "convention" for  these accessory methods that don't really need any attribute of the class. I'm glad to change if you think it's better to have them all as class methods.

Reply:
Hmm, yes, please do it. As they are used only in this class, I think it make more sense.

Reply:
Done!